File: src/TeamCubing.Domain/Models/UserSolve.cs
namespace TeamCubing.Domain.Models;

public class UserSolve
{
    public string RoomName { get; set; }

    public int SolveNumber { get; set; }

    public string Scramble { get; set; }

    public RoomPuzzle Puzzle { get; set; }

    public int Time { get; set; }

    public Penalty Penalty { get; set; }

    public DateTime DateAdded { get; set; }
}

File: src/TeamCubing.API/ClientApp/src/app/models/userSolve.ts
import { RoomPuzzle } from './roomSettings';
import { Penalty } from './solve';

export interface UserSolve {
  roomName: string;
  scramble: string;
  puzzle: RoomPuzzle;
  time: number;
  penalty: Penalty;
  dateAdded: Date;
}
